apps/performance-tests/src/hierarchies/Filtering.test.ts

```diff
@@ -0,0 +1,158 @@
+                        ecInstanceId: { selector: `this.ECInstanceId` },
+                        nodeLabel: { selector: `this.UserLabel` },
+                        extendedData: {
+                          ecInstanceId: { selector: `this.ECInstanceId` },
+                        },
+                      })}
+                      FROM ${fullClassName} AS this
+                      WHERE this.ECInstanceId IN (${parentIdsArr.join(", ")})
+                    `,
+                  },
+                },
+              ];
+            }
+
+            if (parentIdsArr.includes(props.parentNode?.extendedData?.ecInstanceId)) {
+              const query = createNodesQueryClauseFactory({
+                imodelAccess,
+                instanceLabelSelectClauseFactory: createBisInstanceLabelSelectClauseFactory({ classHierarchyInspector: imodelAccess }),
+              });
+              return [
+                {
+                  fullClassName,
+                  query: {
+                    ecsql: `
+                      SELECT ${await query.createSelectClause({
+                        ecClassId: { selector: `this.ECClassId` },
+                        ecInstanceId: { selector: `this.ECInstanceId` },
+                        nodeLabel: { selector: `this.UserLabel` },
+                      })}
+                      FROM ${fullClassName} AS this
+                      WHERE this.[Parent] is NULL AND this.ECInstanceId NOT IN (${startingIndex}, ${parentIdsArr.join(", ")})
+                    `,
+                  },
+                },
+              ];
+            }
+
+            return [];
+          },
+        }),
+        filtering,
+      };
+    },
+    cleanup: (props: { iModel: IModelDb }) => {
+      props.iModel.close();
+    },
+    test: async (props) => {
+      const provider = new StatelessHierarchyProvider(props);
+      const nodeCount = await provider.loadHierarchy();
+      expect(nodeCount).to.eq(totalNumberOfFilteringPaths);
+    },
+  });
+});
+
+function getFilteringPaths(
+  totalNumberOfFilteringPaths: number,
+  parentIdsArr: number[],
+  startingIndex: number,
+  maximumNumberOfPathsForSingleParent: number,
+): HierarchyFilteringPath[] {
+  const { schemaName, itemsPerGroup, defaultClassName } = Datasets.CUSTOM_SCHEMA;
+  const filteringPaths = new Array<HierarchyFilteringPath>();
+  for (let i = 0; i < totalNumberOfFilteringPaths; ++i) {
+    const hundredsPosition = Math.floor(i / itemsPerGroup);
+    const id = `0x${(i + startingIndex).toString(16)}`;
+    const nearestParentId = `0x${parentIdsArr[i % Math.ceil(totalNumberOfFilteringPaths / maximumNumberOfPathsForSingleParent)].toString(16)}`;
+    filteringPaths.push([
+      { className: `${schemaName}.${defaultClassName}_0`, id: `0x${startingIndex.toString(16)}` },
+      { className: `${schemaName}.${defaultClassName}_0`, id: nearestParentId },
+      { className: `${schemaName}.${defaultClassName}_${hundredsPosition}`, id },
+    ]);
+  }
```

Comment:
please add some comments explaining what's happening here

Reply:
Didn't add a comment, just simplified it.